### DISCUSSION TITLE
`launch` configuration for TopDownOrtho camera view

### DISCUSSION THREAD

**Discussion by Simonmaignan** [lucasw/rviz_camera_stream#50] [Q&A]
Hello @lucasw ,

first I'd like to thank you for this really useful tool that I found and started using recently.

I'm still learning how to use properly the tool and I have one question regarding the camera configuration in the `launch` file.
How can I define the `launch` file in order to have the equivalent of the RViz `TopDownOrtho` view?
All the configurations I found in your launch file seem to be the equivalent of the RViz `ThirdPersonFollower` view.

Thanks in advance for your tips

Simon

**Comment by lucasw** [comment]
The best you can do with the current code is make the fx/fy values very large while making the camera origin far from what the camera is pointed at (but there is a far_plane maximum of 100.0, it can't be further from that unless you alter the code)- this only approximates an ortho view.

To make a real ortho view there would need to be something in the input camera_info that puts the rviz camera into an orthographic projection mode instead of perspective- could be a string like `"ortho"` in the `distortion_model` field.

Then in https://github.com/lucasw/rviz_camera_stream/blob/noetic-devel/src/camera_display.cpp#L702-L723 he coefficients of the projection matrix would be set up differently- or possibly Ogre3D has some convenience functions to use, but I think the just the projection matrix values could be changed to accomplish it- https://www.songho.ca/opengl/gl_projectionmatrix.html#ortho may work.

**Reply by Simonmaignan** [reply]
Thanks for the fast answer @lucasw .
I'll give a try to the larger fx/fy value first to check if it's enough for my use case.
I never used `Ogre3D` and have no experience with projection matrices so I'm not sure I'd be able to properly adapt the code in case it's needed.

**Comment by Simonmaignan** [comment]
Here is the launch file I was able to create to stream the equivalent of a top down ortho camera with current code state
```
<group ns="ortho_cam">
    <node pkg="tf2_ros" type="static_transform_publisher" name="camera_broadcaster"
      args="30.0 0.0 30.0 0.0 0.0 -3.14 fixed_tf_on_robot ortho_cam" />
    <node name="camera_info" pkg="rostopic" type="rostopic"
      args="pub /rviz/ortho_cam/camera_info sensor_msgs/CameraInfo
     '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: 'ortho_cam'},
      height: 720, width: 1280, distortion_model: 'plumb_bob',
      D: [0],
      K: [400.0, 0.0, 640, 0.0, 400.0, 360.0, 0.0, 0.0, 1.0],
      R: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0],
      P: [400.0, 0.0, 640, 0.0, 0.0, 400, 360, 0.0, 0.0, 0.0, 1.0, 0.0],
      binning_x: 0, binning_y: 0,
      roi: {x_offset: 0, y_offset: 0, height: 720, width: 1280, do_rectify: false}}' -r 50"
      output="screen" />
  </group>
```
